daml-assistant/daml-project-config/DA/Daml/Project/Consts.hs

```diff
@@ -43,6 +44,11 @@ import DA.Daml.Project.Types
 damlPathEnvVar :: String
 damlPathEnvVar = "DAML_HOME"
 
+-- | The DAML_CACHE environment variable determines the path of the daml
+-- assistant cache directory. This defaults to XDG_CACHE_HOME.
+damlCacheEnvVar :: String
+damlCacheEnvVar = "XDG_CACHE_HOME"
```

Comment:
I disagree with using `XDG_CACHE_HOME` here directly, and it will cause problems since it's not an environment variable "owned" by the assistant.

This should really be `DAML_CACHE`, and the assistant should set up the `DAML_CACHE` environment variable when it runs (using `XDG_CACHE_HOME` as default), the way it already sets up e.g. `DAML_PROJECT` and `DAML_SDK_VERSION` and so on.

```suggestion
damlCacheEnvVar = "DAML_CACHE"
```

Reply:
good catch, this should definitely be set by the assistant.

Reply:
ok, let's do it this way then.